File: Exercicios/ex001.py
"""
Classe Pessoa: Crie uma classe que modele uma pessoa:

Atributos: nome, idade, peso e altura
Métodos: Envelhercer, engordar, emagrecer, crescer. Obs: Por padrão, a 
cada ano que nossa pessoa envelhece, sendo a idade dela menor que 21 anos, ela deve crescer 0,5 cm.
"""

class Pessoa:
    def __init__(self, nome, idade, peso, altura):
        self.nome = nome
        self.idade = idade
        self.peso = peso
        self.altura = altura

    def envelhecer(self, anos):
        try:
            if anos <= 0:
                print('--- Valor inválido ---')
            else:
                    self.idade += anos

                    if self.idade >= 21:
                         self.altura += (anos*0.005)
        except:
            print('--- Não foi possível realizar a ação ---')

    def engordar(self, peso):
        try:

            if peso <= 0:
                print('--- Valor inválido ---')
            self.peso += peso
        
        except:
             print('--- Não foi possível realizar ação ---')

    def emagrecer(self, peso):
        try:
            if peso <= 0:
                print('--- Valor inválido ---')
            self.peso -= peso
        
        except:
             print('--- Não foi possível realizar ação ---')  

    def crescer(self, altura):
        try:
            if altura <= 0:
                print('--- Valor inválido ---')
            self.altura += altura
        
        except:
             print('--- Não foi possível realizar ação ---')  

pessoa1= Pessoa('Vitória', 21, 45, 1.60)

print(pessoa1.__dict__)

pessoa1.crescer(0.05)

print(pessoa1.__dict__)

File: Exercicios/ex001.ts
class PessoaTs {
    nome: string;
    idade: number;
    peso: number;
    altura: number;
  
    constructor(nome: string, idade: number, altura: number, peso: number) {
      this.nome = nome;
      this.idade = idade;
      this.altura = altura;
      this.peso = peso;
    }
  
    envelhecer(idade: number): void {
      try {
        if (idade <= 0) {
          console.log('Valor inválido');
          return;
        }
  
        this.idade += idade;
  
        if (this.idade < 21) {
          this.altura += idade * 0.005;
        }
      } catch (error) {
        console.log('Não foi possível realizar essa ação');
      }
    }
  
    engordar(peso: number): void {
      try {
        if (peso <= 0) {
          console.log('Valor inválido');
          return;
        }
  
        this.peso += peso;
      } catch (error) {
        console.log('Não foi possível realizar essa ação');
      }
    }
  
    emagrecer(peso: number): void {
      try {
        if (peso <= 0) {
          console.log('Valor inválido');
          return;
        }
  
        this.peso -= peso;
      } catch (error) {
        console.log('Não foi possível realizar essa ação');
      }
    }
  
    crescer(altura: number): void {
      try {
        if (altura <= 0) {
          console.log('Valor inválido');
          return;
        }
  
        this.altura += altura;
      } catch (error) {
        console.log('Não foi possível realizar essa ação');
      }
    }
  }
  
  const pessoa1Ts = new PessoaTs('Vitória', 16, 45, 1.60);
  
  pessoa1Ts.envelhecer(3);
  
  console.log(pessoa1Ts.idade);
  